tests/platforms/web/static/test_build__process_wheel.py

```diff
@@ -93,9 +82,267 @@ def test_process_wheel_no_content(build_command, tmp_path):
+
+    assert inserts == {}
+
+
+def test_process_wheel_deploy_inserts(build_command, tmp_path):
+    """Deploy inserts are collected into the correct insert slot."""
+    wheel_filename = create_wheel(
+        tmp_path,
+        extra_content=[
+            ("dummy/deploy/inserts/index.html~header", "<script>alert('hi')</script>"),
+            (
+                "dummy/deploy/inserts/static/css/briefcase.css~CSS",
+                "body { margin: 0; }",
+            ),
+        ],
+    )
+
+    inserts = {}
+    build_command._process_wheel(wheelfile=wheel_filename, inserts=inserts)
+
+    # The index.html header insert exists
+    assert "index.html" in inserts
+    assert "header" in inserts["index.html"]
+    contribs = inserts["index.html"]["header"]
+    assert any("<script>" in v for v in contribs.values())
+
+    # The CSS insert exists
+    assert "static/css/briefcase.css" in inserts
+    assert "CSS" in inserts["static/css/briefcase.css"]
+    css_contribs = inserts["static/css/briefcase.css"]["CSS"]
+    assert any("body { margin: 0; }" in v for v in css_contribs.values())
+
+
+def test_process_wheel_legacy_css_warning_once(build_command, tmp_path, monkeypatch):
+    """Legacy CSS files trigger a single deprecation warning."""
+    wheel_filename = create_wheel(
+        tmp_path,
+        extra_content=[
+            ("dummy/static/one.css", "h1 { color: blue; }"),
+            ("dummy/static/two.css", "h2 { color: green; }"),
+        ],
+    )
+
+    # Check on console.warning to count legacy warnings
+    warnings = []
+    monkeypatch.setattr(
+        build_command.console, "warning", lambda msg: warnings.append(msg)
+    )
+
+    inserts = {}
+    build_command._process_wheel(wheelfile=wheel_filename, inserts=inserts)
+
+    legacy_msgs = [m for m in warnings if "legacy '/static' CSS detected" in m]
+    assert len(legacy_msgs) == 1
+
+
+def test_process_wheel_non_utf8_insert(build_command, tmp_path):
+    """Non-UTF8 deploy insert raises an UnicodeDecodeError."""
+    bad = b"\xff\xfe\xfa"
+    wheel_path = Path(tmp_path) / "dummy-1.2.3-py3-none-any.whl"
+    with zipfile.ZipFile(wheel_path, "w") as zf:
+        zf.writestr("dummy/deploy/inserts/index.html~header", bad)
+
+    inserts = {}
+    with pytest.raises(UnicodeDecodeError):
+        build_command._process_wheel(wheelfile=wheel_path, inserts=inserts)
+
+
+def test_process_wheel_non_utf8_legacy_css(build_command, tmp_path):
+    """Non-UTF8 legacy CSS raises an UnicodeDecodeError."""
+    bad = b"\xff\xfe\xfa"
+    wheel_path = Path(tmp_path) / "dummy-1.2.3-py3-none-any.whl"
+    with zipfile.ZipFile(wheel_path, "w") as zf:
+        zf.writestr("dummy/static/bad.css", bad)
+
+    inserts = {}
+    with pytest.raises(UnicodeDecodeError):
+        build_command._process_wheel(wheelfile=wheel_path, inserts=inserts)
+
+
+def test_handle_legacy_css_warning_once_and_append(
+    build_command, tmp_path, monkeypatch
+):
```

Comment:
Two notes here:
1. As this method is testing `handle_legacy_css`, it should be in a `test_build_handle_legacy_css.py` file
2. When black/ruff splits the argument list breaks onto a new line, our preference is to move to "one argument per line" format.